src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -267,12 +277,38 @@ let run (conf : Scan_CLI.conf) : Exit_code.t =
                 Some (file_match_results_hook conf filtered_rules) )
           | { output_format; _ } -> (output_format, None)
         in
+        Profiler.save profiler ~name:"core_time";
```

Comment:
I think it would be better to group those 2 calls in one if it's possible, for example
Profile.execute_and_profile ~name:"core_time" (fun () -> ...)

Reply:
when it's possible hwen the 2 calls are in the same scope

Reply:
or at least a Profiler.start and Profiler.end

Reply:
(or better, both)